scripts/exrtma3d_gsianl.ksh

```diff
@@ -148,60 +153,141 @@ fi
+  ${LN} -s ${DATAOBSHOME}/rtma_ru.t${HH}${subcyc}z.lghtng.tm00.bufr_d ./lghtInGSI
+elif [ -r "${DATAOBSHOME}/${RUN}.t${HH}${subcyc}z.LightningInGSI_bufr.bufr" ]; then
+  ${LN} -s ${DATAOBSHOME}/${RUN}.t${HH}${subcyc}z.LightningInGSI_bufr.bufr ./lghtInGSI
 else
   ${ECHO} "Warning: ${DATAOBSHOME}: LightningInGSI.bufr does not exist!"
 fi
 
+
 if [ -r "${DATAOBSHOME}/NASALaRCCloudInGSI.bufr" ]; then
   ${LN} -s ${DATAOBSHOME}/NASALaRCCloudInGSI.bufr ./larcInGSI
-elif [ -r "${DATAOBSHOME}/hrrr.t${HH}z.NASALaRCCloudInGSI.bufr" ]; then
-  ${LN} -s ${DATAOBSHOME}/hrrr.t${HH}z.NASALaRCCloudInGSI.bufr ./larcInGSI
-elif [ -r "${DATAOBSHOME}/${RUN}.t${HH}z.NASALaRCCloudInGSI.bufr" ]; then
-  ${LN} -s ${DATAOBSHOME}/${RUN}.t${HH}z.NASALaRCCloudInGSI.bufr ./larcInGSI
+elif [ -r "${DATAOBSHOME}/hrrr.t${HH}${subcyc}z.NASALaRCCloudInGSI.bufr" ]; then
+  ${LN} -s ${DATAOBSHOME}/hrrr.t${HH}${subcyc}z.NASALaRCCloudInGSI.bufr ./larcInGSI
+elif [ -r "${DATAOBSHOME}/rtma_ru.t${HH}${subcyc}z.lgycld.tm00.bufr_d" ]; then
+  ${LN} -s ${DATAOBSHOME}/rtma_ru.t${HH}${subcyc}z.lgycld.tm00.bufr_d ./larcInGSI
+elif [ -r "${DATAOBSHOME}/${RUN}.t${HH}${subcyc}z.NASALaRCCloudInGSI.bufr" ]; then
+  ${LN} -s ${DATAOBSHOME}/${RUN}.t${HH}${subcyc}z.NASALaRCCloudInGSI.bufr ./larcInGSI
 else
   ${ECHO} "Warning: ${DATAOBSHOME}: NASALaRCCloudInGSI.bufr does not exist!"
 fi
 
-# Link statellite radiance data
+# Set runtime and save directories
+export endianness=Big_Endian
+
+# Set variables used in script
+#   ncp is cp replacement, currently keep as /bin/cp
+ncp=/bin/cp
+
+export HYB_ENS=".true."
+
+# Get Fv3GDAS Enkf files
+# We expect 80 total files to be present (80 enkf)
+export nens=80
+
+# Not using FGAT or 4DEnVar, so hardwire nhr_assimilation to 3
+export nhr_assimilation=03
+##typeset -Z2 nhr_assimilation
+
 
-# Link the radial velocity data
+python ${UTILrtma3d_dev}/getbest_EnKF_FV3GDAS.py -v $YYYYMMDDHH --exact=no --minsize=${nens} -d ${COMINGDAS}/enkfgdas -m no -o filelist${nhr_assimilation} --o3fname=gfs_sigf${nhr_assimilation} --gfs_nemsio=yes
+ 
 
-## 
-## Find closest GFS EnKF forecast to analysis time
-##
-## 
-## Link to pre-processed GFS EnKF forecast members
-##
+#Check to see if ensembles were found 
+numfiles=`cat filelist03 | wc -l`
+
+if [ $numfiles -ne 80 ]; then
+  echo "Ensembles not found - turning off ifhyb!"
+  export ifhyb=".false."
+else
+#   we have 80 files, figure out if they are all the right size
+#   if not, set ifhyb=false
+    cp ${UTILrtma3d_dev}/convert.sh .
+    ${UTILrtma3d_dev}/check_enkf_size.sh
+fi
+
+if [ ${HRRRDAS_BEC} -gt 0 ]; then
+  ${ECHO} "\$HRRRDAS_BEC=${HRRRDAS_BEC}, so HRRRDAS will be used if available"
+  #----------------------------------------------------
+  # generate list of HRRRDAS members for ensemble covariances
+  # Use 1-hr forecasts from the HRRRDAS cyclin
+  c=1
+   ${LS} ${COMINhrrrdas}/hrrr.t${HH}00z.f0100.mem000${c}.netcdf > filelist.hrrrdas
+
```

Comment:
Found a bug here:

First ensemble member  will NOT be linked to wrf_en001.
Need to add one line here for c=1:
${LN} -sf  ${COMINhrrrdas}/hrrr.t${HH}00z.f0100.mem0001.netcdf    wrf_en001

or re-design the while-do block and the embedded if-else block for c=1, and c>1.